src/btree/bt_prefetch.c

```diff
@@ -89,13 +89,15 @@ __wt_prefetch_page_in(WT_SESSION_IMPL *session, WT_PREFETCH_QUEUE_ENTRY *pe)
 
     WT_STAT_CONN_INCR(session, block_prefetch_pages_read);
 
+    WT_ENTER_GENERATION(session, WT_GEN_SPLIT);
     if (__wt_ref_addr_copy(session, pe->ref, &addr)) {
         WT_ERR(__wt_page_in(session, pe->ref, WT_READ_PREFETCH));
         WT_ERR(__wt_page_release(session, pe->ref, 0));
     } else
-        return (WT_ERROR);
+        ret = (WT_ERROR);
```

Comment:
I wonder if the original code here had a bug? This changes the behaviour slightly as now we'll call `__wt_page_release` and return `WT_ERROR` whereas previously we would just return `WT_ERROR`. What do you think Hari?

Reply:
@mm-ng

Reply:
Ok, so the code you have here is outdated. We've changed it back to `return (WT_ERROR)` rather than `ret = (WT_ERROR)`.

Reply:
I merged the latest develop code that does not have another page release function call.